sockeye/inference.py

```diff
@@ -962,36 +1005,42 @@ def _make_result(self,
         :param translation: The translation and score.
         :return: TranslatorOutput.
         """
-        target_ids = translation.target_ids
-        target_tokens = [self.vocab_target_inv[target_id] for target_id in target_ids]
-        target_string = C.TOKEN_SEPARATOR.join(data_io.ids2tokens(target_ids, self.vocab_target_inv, self.strip_ids))
+        primary_target_tokens, primary_translation, other_target_tokens, other_target_strings = \
+            self._get_translation_tokens_and_factors(translation.target_ids)
 
         if translation.nbest_translations is None:
+            primary_target_tokens, primary_translation, other_target_tokens, other_target_strings = \
```

Comment:
Isn't this a duplicate to the line before the `if` statement?

Reply:
True its a bit redundant. I moved the above line into the lower else-block. Can you review this carefully? This is quite complicated unfortunately.

Reply:
It could be moved outside of the if block even couldn't it?

Reply:
yes, will change